### ISSUE TITLE
Get problem with --fp16 in BERT nan loss

### ISSUE THREAD

**Issue by AoqunJin** [microsoft/varuna#18]
########################################################
python -m varuna.run_varuna --machine_list ./ips.txt --gpus_per_node 4 \
--batch_size 4096 --nstages 2 \
--chunk_size 16 \
--code_dir /root/code/space/DeepLearningExamples/PyTorch/LanguageModeling/BERT run_pretraining.py \
--input_dir /root/code/dataset/BERT/hdf5_lower_case_1_seq_len_128_max_pred_20_masked_lm_prob_0.15_random_seed_12345_dupe_factor_5/wikicorpus_en \
--config_file /root/code/space/DeepLearningExamples/PyTorch/LanguageModeling/BERT/bert_config.json \
--output_dir /root/code/space/DeepLearningExamples/PyTorch/LanguageModeling/BERT/out \
--do_train --fp16 --max_steps 10 --varuna 

########################################################
0 Overflow !!2 Overflow !!

1 Overflow !!3 Overflow !!

032 : update_scale(): _has_overflow, dynamic. _loss_scale =   : update_scale(): _has_overflow, dynamic. _loss_scale =   : update_scale(): _has_overflow, dynamic. _loss_scale =  524288.0
524288.0
524288.01
 : update_scale(): _has_overflow, dynamic. _loss_scale =  524288.0
Could not send progress update messageCould not send progress update messageCould not send progress update message

Could not send progress update message
DLL 2023-06-15 20:14:36.849959 - Training Epoch: 0 Training Iteration: 1  average_loss : 11.3046875  step_loss : 11.3046875  learning_rate : 4.743416490252569e-05 
3 Overflow !!
1 Overflow !!
2 Overflow !!
0 Overflow !!
13  : update_scale(): _has_overflow, dynamic. _loss_scale = : update_scale(): _has_overflow, dynamic. _loss_scale =  0  262144.0262144.0: update_scale(): _has_overflow, dynamic. _loss_scale = 

 262144.0
2 : update_scale(): _has_overflow, dynamic. _loss_scale =  262144.0
Could not send progress update messageCould not send progress update messageCould not send progress update message

Could not send progress update message
DLL 2023-06-15 20:14:37.136949 - Training Epoch: 0 Training Iteration: 2  average_loss : nan  step_loss : nan  learning_rate : 4.4721359549995795e-05 
0 Overflow !!2 Overflow !!

3 Overflow !!
1 Overflow !!
1 0: update_scale(): _has_overflow, dynamic. _loss_scale = 3  : update_scale(): _has_overflow, dynamic. _loss_scale =   131072.0: update_scale(): _has_overflow, dynamic. _loss_scale =  
131072.0131072.0
```[tasklist]
### Tasks
```

**Comment by AoqunJin** [comment]
add warmup_proportion to solve this.
